lib/galaxy/tools/imp_exp/__init__.py

```diff
@@ -74,6 +74,11 @@ def get_tag_str( tag, value ):
                 #
                 # Create history.
                 #
+                if not os.path.exists(os.path.join( archive_dir, 'history_attrs.txt')):
+                    dirs = os.listdir(archive_dir)
+                    for d in dirs:
+                        if os.path.isdir(os.path.join(archive_dir, d)):
+                                archive_dir = os.path.join(archive_dir, d)
```

Comment:
The indentation is off here, it should be 4 spaces always.
Also I think you can add a break after this, since we expect only load a single `archive_dir`.